k8s/k8s.go

```diff
@@ -250,6 +263,118 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 	return result, nil
 }
 
+// ipAddrsResult parses the ipAddrs annotation and calls IPAM plugin for
+// each IP passed to it by setting CNI_ARGS IP=ip_addr, and returns the type.Result.
+// Example annotation value string: "[\"10.0.0.1\", \"2001:db8::1\"]"
+func ipAddrsResult(ipAddrs string, conf utils.NetConf, args *skel.CmdArgs, logger *log.Entry) (*types.Result, error) {
+	var ips []string
+
+	err := json.Unmarshal([]byte(ipAddrs), &ips)
+	if err != nil {
+		logger.WithField("annotation", ipAddrs).Error("Failed to parse ipAddrs as json")
+		return nil, fmt.Errorf("Failed to parse ipAddrs: %s", err)
+	}
+
+	result := types.Result{}
+
+	// annotation value can't be empty.
+	if len(ips) == 0 {
+		logger.WithField("annotation", "cni.projectcalico.org/ipAddrs").Error("No IPs specified")
+		return nil, fmt.Errorf("No IPs specified with annotation \"cni.projectcalico.org/ipAddrs\"")
```

Comment:
This would be more clear if it was worded like `"Annotation \"cni.projectcalico.org/ipAddrs\" specified but empty"`